### PR TITLE
MM-12398/MM-13153 Fix typing quickly in autocomplete selecting wrong item and fix channel switcher

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
When typing really quickly in the channel switcher and pressing enter, you could be sent to a channel that does not match what you have typed. This fixes that for all autocomplete by checking if the selection matches what has been typed. If it doesn't, then we go fetch the latest results before completing.

Also fixed MM-13153 because that was blocking testing on this.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12398
https://mattermost.atlassian.net/browse/MM-13153

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: components/quick_switch_modal/quick_switch_modal.jsx [hunk 1]
```diff
@@ -111,7 +111,7 @@ export default class QuickSwitchModal extends React.PureComponent {
 
         if (this.state.mode === CHANNEL_MODE) {
             const selectedChannel = selected.channel;
-            this.props.actions.switchToChannelById(selectedChannel.id).then((result) => {
+            this.props.actions.switchToChannel(selectedChannel).then((result) => {
```
Comment (jwilander) [inline components/quick_switch_modal/quick_switch_modal.jsx]: This is the fix for MM-13153

Comment (enahum) [reply components/quick_switch_modal/quick_switch_modal.jsx]: Curious to know why was this changed to `switchToChannelById` after we changed this to `switchToChannel` when we did the migration to redux? I only ask cause is being reverted once more to be `switchToChannel`

Comment (jwilander) [reply components/quick_switch_modal/quick_switch_modal.jsx]: Oh looks like this did it to fix GMs https://github.com/mattermost/mattermost-webapp/pull/2047/files

I should check that case and fix it if need be

Comment (enahum) [reply components/quick_switch_modal/quick_switch_modal.jsx]: sounds good, add me to re-review it if needed

File: actions/views/channel.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ import {getLastViewedChannelName} from 'selectors/local_storage';
 import {browserHistory} from 'utils/browser_history';
 import {ActionTypes} from 'utils/constants.jsx';
 import {isMobile} from 'utils/utils.jsx';
+import Constants from '../../utils/constants';
```
Comment (enahum) [inline actions/views/channel.js]: isn't this a dup import from two lines above? not that is blocker but you could use
```js
import {General} from 'mattermost-redux/constants'

channel.type === General.DM_CHANNEL
```

Comment (jwilander) [reply actions/views/channel.js]: Ahh that's my auto-importer doing that. Let me fix it

**Pull request merged**